monkey/infection_monkey/monkey.py

```diff
@@ -369,13 +370,14 @@ def try_exploiting(self, machine, exploiter):
             exploiter.send_exploit_telemetry(result)
         return False
 
-    def successfully_exploited(self, machine, exploiter):
+    def successfully_exploited(self, machine, exploiter, RUNS_AGENT_ON_SUCCESS=True):
         """
         Workflow of registering successfully exploited machine
         :param machine: machine that was exploited
         :param exploiter: exploiter that succeeded
         """
-        self._exploited_machines.add(machine)
+        if RUNS_AGENT_ON_SUCCESS:
+            self._exploited_machines.add(machine)
```

Comment:
Does the machine appear exploited in the map and report?

Reply:
Yes